### DISCUSSION TITLE
System freezes when running tests.

### DISCUSSION THREAD

**Discussion by gokullan** [ToolJet/ToolJet#6813] [Q&A]
Hi, there!
I use a laptop running Linux Mint on Intel i5-10210U with 8 GB DDR4 RAM . When I run `npm run --prefix server test`, the system freezes after a few seconds. I tried running the tests one by one with  `npm run --prefix server test filename` (where filename refers to files in `ToolJet/server/test/services`) and the tests run, but I get a message at the end saying `Jest did not exit one second after the test run has completed.` Is the freeze because of my laptop-spec limitations? Or is the issue something else? How do I ensure the tests run properly?

**Comment by habeeb-an** [comment]
I also have almost same spec
my testing also freezes
is there any way to not test manually and do pull request then automatically tests?
also I did try to run tests using cypress-tests but the even the automatic login failed with tooljetio email
I didn't touch those parts

**Reply by habeeb-an** [reply]
@gokullan hope you would let me know when you got the fix

**Reply by gokullan** [reply]
Yes, sure. @habeeb-an, you too let me know if you find any workaround.

**Comment by adishM98** [comment]
Hi @gokullan @habeeb-an, 

You can push the code and add the label `run-ci` which will run test action. Once that is successful the PR will be reviewed.

**Reply by gokullan** [reply]
Cool, thanks @adishM98 👍

**Reply by habeeb-an** [reply]
@adishM98 i can't see any option to add ```run-ci``` label

**Reply by gokullan** [reply]
@adishM98 I don't think we have permissions to add labels.
